SimG4Core/SensitiveDetector/interface/SensitiveDetectorMakerBase.h

```diff
@@ -47,8 +47,8 @@ class SensitiveDetectorMakerBase
 			const edm::ParameterSet& p,
 			const SimTrackManager* m,
 			SimActivityRegistry& reg,
-			std::auto_ptr<SensitiveTkDetector>& oTK,
-			std::auto_ptr<SensitiveCaloDetector>& oCalo) const =0;
+			std::unique_ptr<SensitiveTkDetector>& oTK,
+			std::unique_ptr<SensitiveCaloDetector>& oCalo) const =0;
```

Comment:
passing a reference to a unique pointer without transferring ownership may cause trouble if an owner decides to reset it. Would a shared_ptr be safer here?